lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -840,3 +842,41 @@ def get_date_from_repeat_instance(instance_id: int) -> str:
     Returns the date as a string because that is how it is used.
     """
     return (STUDY_START_DATE + relativedelta(days=(instance_id -1))).strftime('%Y-%m-%d')
+
+
+def insert_range_of_synthetic_enrollment_dets(start_record_id: int, end_record_id: int,
+        generated_by: str) -> None:
+    """
+    Inserts synthetic DET records into the `receiving.redcap_det` table to mimic
+    changes to the Enrollment Questionnaire for the range of record IDs (inclusive)
+    specified.
+    Requires the `REDCAP_API_TOKEN_{REDCap URL}_{REDCap project ID}` environment variable be set.
+    """
+    project = Project(REDCAP_URL, PROJECTS[0].id, dry_run = False)
+
+    record = {
+        "project" : project,
+        "enrollment_questionnaire_complete" : "", # Don't set a value
+        "redcap_event_name" : ENROLLMENT_EVENT_NAME,
+        }
+
+    current_record_id = start_record_id
+    dets = []
+
+    while current_record_id <= end_record_id:
+        record['record_id'] = str(current_record_id)
+        dets.append(det(project, record, 'enrollment_questionnaire', generated_by))
+        current_record_id += 1
+
+    dets_json = [
+        (Json(d),)
+            for d in dets ]
```

Comment:
Since we've been bumping into memory issues, I might recommend creating only one `dets` list (instead of `dets` and `dets_json`), and converting to `Json` while appending to the `dets` list as is done in `offer_uw_testing`.

If you've broken up the DET + conversion to JSON steps separately for better readability, another approach could be to convert the det to JSON then append it in the following line. e.g.
```py
        current_det = det(project, record, 'enrollment_questionnaire', generated_by)
        dets.append(Json(current_det))
```